### PR TITLE
API. No auction response

### PR DESCRIPTION
**Pull request by yoalex5**
GitHub issue #122 

1. PREBID_SERVER_ERROR was made as default error
2. Unit tests were modified

### CODE REVIEW

File: PrebidMobile/API1.0/src/test/java/org/prebid/mobile/PrebidServerAdapterTest.java [hunk 1]
```diff
@@ -112,7 +115,13 @@ public void testInvalidPrebidServerAccountIdForRubiconHostedPrebidServer() {
         adapter.requestDemand(requestParams, mockListener, uuid);
         Robolectric.flushBackgroundThreadScheduler();
         Robolectric.flushForegroundThreadScheduler();
-        verify(mockListener).onDemandFailed(ResultCode.INVALID_ACCOUNT_ID, uuid);
+
+        ArgumentCaptor<ResultCode> resultCodeAC = ArgumentCaptor.forClass(ResultCode.class);
+        ArgumentCaptor<String> uuidAC = ArgumentCaptor.forClass(String.class);
+
+        verify(mockListener).onDemandFailed(resultCodeAC.capture(), uuidAC.capture());
+        assertThat(resultCodeAC.getValue(), Matchers.either(Matchers.is(ResultCode.INVALID_ACCOUNT_ID)).or(Matchers.is(ResultCode.PREBID_SERVER_ERROR)));
```
[the remark this answers is not in the document]
Comment (yoalex5) [reply PrebidMobile/API1.0/src/test/java/org/prebid/mobile/PrebidServerAdapterTest.java]: I have added a separate unit test

**Comment by yoalex5** [comment]
@avohraa, could you please approve this PR? 
I have just merged the master to the current branch

**Comment by yoalex5** [comment]
@bszekely1, will you please review this PR. All the changes were approved by @avohraa, I just merged master into this branch. After that I will be unblocked

File: PrebidMobile/API1.0/src/test/java/org/prebid/mobile/PrebidServerAdapterTest.java (hunk 1 above)
Comment (avohraa) [inline PrebidMobile/API1.0/src/test/java/org/prebid/mobile/PrebidServerAdapterTest.java]: What are the cases that enforce us to assert two ResultCode?
- INVALID_ACCOUNT_ID
- PREBID_SERVER_ERROR

This test case is for the INVALID_ACCOUNT_ID, then why should it pass if onDemandFailed is obtained with other ResultCode (PREBID_SERVER_ERROR in this case)

Also, can we mock the response instead? Something like this:
`server.enqueue(new MockResponse().setResponseCode(400).setBody("Invalid request format: No stored request found for id: 1001_INVALID_ACCOUNT_ID"));`

File: PrebidMobile/API1.0/src/test/java/org/prebid/mobile/PrebidServerAdapterTest.java [hunk 2]
```diff
@@ -148,7 +157,13 @@ public void testInvalidPrebidServerConfigIdForRubiconHostedPrebidServer() {
         adapter.requestDemand(requestParams, mockListener, uuid);
         Robolectric.flushBackgroundThreadScheduler();
         Robolectric.flushForegroundThreadScheduler();
-        verify(mockListener).onDemandFailed(ResultCode.INVALID_CONFIG_ID, uuid);
+
+        ArgumentCaptor<ResultCode> resultCodeAC = ArgumentCaptor.forClass(ResultCode.class);
+        ArgumentCaptor<String> uuidAC = ArgumentCaptor.forClass(String.class);
+
+        verify(mockListener).onDemandFailed(resultCodeAC.capture(), uuidAC.capture());
+        assertThat(resultCodeAC.getValue(), Matchers.either(Matchers.is(ResultCode.INVALID_CONFIG_ID)).or(Matchers.is(ResultCode.PREBID_SERVER_ERROR)));
```
Comment (avohraa) [inline PrebidMobile/API1.0/src/test/java/org/prebid/mobile/PrebidServerAdapterTest.java]: Can we make similar changes for INVALID_CONFIG_ID as well?

Comment (yoalex5) [reply PrebidMobile/API1.0/src/test/java/org/prebid/mobile/PrebidServerAdapterTest.java]: @avohraa, I have updated the PR

**Comment by yoalex5** [comment]
@avohraa, could you please review these changes

File: PrebidMobile/API1.0/src/test/java/org/prebid/mobile/PrebidServerAdapterTest.java [hunk 3]
```diff
@@ -135,7 +144,15 @@ public void testInvalidPrebidServerConfigIdForAppNexusHostedPrebidServer() {
 
     @Test
     public void testInvalidPrebidServerConfigIdForRubiconHostedPrebidServer() {
-        PrebidMobile.setPrebidServerHost(Host.RUBICON);
+
+        if (!successfulMockServerStarted) {
+            fail("Mock server was not started");
+        }
+        server.enqueue(new MockResponse().setResponseCode(400).setBody("Invalid request format: No stored imp found for id: 1001-1_INVALID_CONFIG_ID"));
```
Comment (avohraa) [inline PrebidMobile/API1.0/src/test/java/org/prebid/mobile/PrebidServerAdapterTest.java]: Sorry for reiterating on this, but can we have that response body from the `MockPrebidServerResponses`?

Something like: `MockPrebidServerResponses.invalidConfigIdFromRubicon`.

File: PrebidMobile/API1.0/src/test/java/org/prebid/mobile/PrebidServerAdapterTest.java [hunk 4]
```diff
@@ -99,7 +99,14 @@ public void testInvalidPrebidServerAccountIdForAppNexusHostedPrebidServer() {
 
     @Test
     public void testInvalidPrebidServerAccountIdForRubiconHostedPrebidServer() {
-        PrebidMobile.setPrebidServerHost(Host.RUBICON);
+        if (!successfulMockServerStarted) {
+            fail("Mock server was not started");
+        }
+        server.enqueue(new MockResponse().setResponseCode(400).setBody("Invalid request format: No stored request found for id: 1001_INVALID_ACCOUNT_ID"));
```
Comment (avohraa) [inline PrebidMobile/API1.0/src/test/java/org/prebid/mobile/PrebidServerAdapterTest.java]: For Invalid Account Id as well.

**Comment by yoalex5** [comment]
@avohraa, done

**Pull request merged**